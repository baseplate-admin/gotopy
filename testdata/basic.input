package test

var GlobInt int = 2
var GlobStr = "a string"

// A struct definition
type MyStru struct {
	A int     // field desc
	B float32 `desc:"field tag"`
	C string  `desc:"more tags"`
}

// A global function
func GlobFun(a, b float32) float32 {
	if a > b && a == 0 || b == 0 {
		return a + b
	} else if a == b {
		return a * b
	} else {
		return a - b
	}
}

// MethOne does something
func (st *MyStru) MethOne(arg1 float32) int {
	rv := st.A
	for a := 0; a < 100; a++ {
		rv += a
	}
	return rv
}

// MethTwo does something
// it is pretty cool
// not really sure about that
func (st *MyStru) MethTwo(arg1, arg2 float32, arg3 int) int {
	rv := st.A
	for a := 0; a < 100; a++ {
		rv += a
	}
	return rv
}
