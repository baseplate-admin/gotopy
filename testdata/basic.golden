package test

GlobInt = 2
GlobStr = "a string"

class MyStru :
# A struct definition
	def __init__(self):
	# field desc
		self.A int
		self.B float32 # `desc:"field tag"`
		self.C string  # `desc:"more tags"`

# MethOne does something
	def MethOne(st, arg1 ):
		"""
		MethOne does something
		"""
		rv = st.A
		for a = 0; a < 100; a++ :
			rv += a

		return rv
	def MethTwo(st, arg1, arg2 , arg3 ):
		"""
		MethTwo does something
		it is pretty cool
		not really sure about that
		"""
		rv = st.A
		for a = 0; a < 100; a++ :
			rv += a

		return rv

# A global function
def GlobFun(a, b ):
	"""
	A global function
	"""
	if a > b and a == 0 or b == 0 :
		return a + b
	else if a == b :
		return a * b
	else :
		return a - b

		
		
		
		
		
